Extension/src/LanguageServer/client.ts

```diff
@@ -2153,11 +2155,13 @@ export class DefaultClient implements Client {
     public cancelReferences(): void {
         referencesParams = null;
         renamePending = false;
-        let cancelling: boolean = referencesPendingCancellations.length > 0;
-        if (!cancelling) {
-            referencesPendingCancellations.push({ reject: () => {}, callback: () => {} });
-            this.languageClient.sendNotification(CancelReferencesNotification);
-            this.references.closeRenameUI();
+        if (referencesRequestPending) {
```

Comment:
By adding this check back in, you are actually undoing a previous fix made by Sean.  Cancel needs to go through even if there is no pending request, for FAR after a preview.  (In this state, the request is no longer pending, rather when final results arrive or another preview is request, it reissues the request).

Reply:
I'll revert this change.

Reply:
Concluded that this flag is needed to address subsequent renames to work. Need to create a different solution for FAR preview.

Reply:
- Counter from [#4344](https://github.com/microsoft/vscode-cpptools/pull/4344) was added to this changeset.
- Added flag to keep track of symbolSearchInProgress 

Some issues still rename though:

- Rename does not cancel if a file was edited when rename is pending (that is rename UI is shown waiting for user to take action).

Overall, canceling rename correctly still needs work.